src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.slc.services;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.model.LetterPrintStatus;
+import uk.gov.hmcts.reform.slc.services.steps.getpdf.FileNameHelper;
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.ParsedReport;
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.Report;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+
+@Component
+public class ReportParser {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportParser.class);
+
+    public ParsedReport parse(Report report) {
+        try (CSVParser parser = parserFor(report.content)) {
+
+            List<LetterPrintStatus> statuses =
+                stream(parser.spliterator(), false)
+                    .map(row -> toPrintStatus(row))
+                    .filter(Objects::nonNull)
+                    .collect(toList());
+
+            return new ParsedReport(report.path, statuses);
+
+        } catch (IOException exc) {
+            throw new ReportParsingException(exc);
+        }
+    }
+
+    private CSVParser parserFor(byte[] csv) throws IOException {
+        return CSVFormat
+            .DEFAULT
+            .withHeader()
+            .parse(new InputStreamReader(new ByteArrayInputStream(csv)));
+    }
+
+    /**
+     * Converts cvs row into a letter print status object.
+     * Returns null if conversion fails.
+     */
+    private LetterPrintStatus toPrintStatus(CSVRecord record) {
+        try {
+            return new LetterPrintStatus(
+                FileNameHelper.extractId(record.get("Filename")),
```

Comment:
can this extract `UUID` instead? i think i saw `UUID.fromString` earlier in the PR. This can be caught early and returned as null from this try/catch block rather than failing to update the state

Reply:
This is now done, LetterPrintStatus uses UUID.